* notes

Target audience: computional scientist and bioinformatician

Objective of the the presentation: convince the audience that SCP is a
promosing field. Attract single-cell data analysts to proteomics, 
attract proteomics analysts to single-cell. 

Level of depth: intermediate. Given the audience and the objective, 
the presentation will avoid technical details, but provide precise
concpetual information supported by real data examples. 

This work/field is important because SCP can unravel unexplored 
biolgical processes that occur at the protein level. SCP directly
inherits from the advantages of the scRNA-Seq field: explore cellular
heterogeneity by minimizing prior knowledge about population composition
(= unbiased). 

- intro: 
    - Start with one or two example research questions and explain how
      single-cell technology can answer them
    - Explain the advantge of SCP over other existing why developing 
      proteomics.
- main: current state of SCP field
    - Methodological workflow
    - Computational workflow
    - Challenges
    - scp and spcdata
- take home:
    - Exciting challenges: model/correct batch effect and missing data
    - Many opportunities: method development, workflow development, benchmarking,...
    - Important refs
    - Get in touch

* Plan

** title slide

Computational challenges in mass spectrometry-based single-cell proteomics

Link to slides

Christophe Vanderaa, Laurent Gatto

** Single-cell technologies (1)

Single-cell technology unravel cellular heterogeneity: 

- Population identification
- Trajectory analysis

Use example of scRNA-Seq data from embryonic liver

** RNA vs protein (2)

- Conceptually: RNA = intention; protein = function. Good enough?
- No: poor correlation between RNA and protein, but RNA easier?
- Yes because RNA amplification, but protein have more copies than RNA

** Single-cell proteomics

Emerged thanks to recent technical advances

Show SCoPE2 method workflow.

** What about computational workflows?

Show SCoPE2 computational workflow: many steps are bulk and the 
workflow is not adapted to single-cell. 

** Downstream analyses

Dimension reduction: specht

Differential analysis: liang

Trajectory analysis: schoof


** Computational challenges 

- Batch effect 
- Missingness 

The two are not independent and need to be accounted for 
simultaneously.

How to get started?

** Performing SCP data

- QFeatures + scp: package to maniuplate and visualize single-cell
  proteomics data
- scpdata: collection of published data ready to be analyzed
- SCP.replication: website with replication of published analyses

Python: SCeptre

** Take home messages

- The era of single-cell proteomics is rising
- Exciting challenges: model/correct batch effect and missing data
- Many opportunities: method development, workflow development, benchmarking,...

Get started: Vanderaa, Slavov, Kelly, 


** Thanks for your attention

Get in touch: 
    - Twitter
    - Github

Mention funding

Link to slides






* Transcript

** slide 1

I'll be presenting our latest work on the principled and reproducible
analysis of mass spectrometry-based single cell proteomics data.

** slide 2

I'm sure that it is abundantly clear for everyone here that single
cell-level assays are of prime importance to study and understand
biology in all its glory. And there's no way we can understand biology
at this level of detail without dedicated and trustworthy statistical
and computational tools.

** slide 3

Whenever a new technology emerges, the first expectation is that it
will inevitably lead to ground breaking discoveries.

But the reality is that there are many things that can go wrong along
the way, that will inevitably lead to false discoveries.

** slide 4

We offer the =scp= Bioconductor package to track, understand and
tackle the things that make single cell data analysis challenging and
interesting. In the next couple of slides, I'll briefly present some
practical aspects of =scp=, its usage and its underlying design, and
how it was used to replicate the SCoPE2 data analysis and better
understand single cell proteomics data analysis challenges.

** slide 5

=scp= is based on the =QFeatures= package to handle multi-level
quantitative data. A complete =QFeatures= dataset contains many
individual assays, including PSM-level quantitation data, that are
joined and aggregated into peptide-level data, that are themselves
aggregated into protein-level quantitation values. All these
operations are recorded, which allow easy navigation between low- and
high-level quantitative features. The experimental design, i.e. the
sample metadata is also stored as part of this multi-assay data
structure.

** slide 6

This slide illustrates the =scp= R syntax, which will look very
familiar to any R and tidyverse user. On the left hand side, you'll
recognise well-known steps in proteomics data processing. These steps
are implemented as specific functions that are chained one to another
using this pipe operator. The code on the right effectively loads the
data, removes low quality runs, performs some QC on the features,
aggregates PSMs into peptides, ... and so on. A user thus combines the
required actions as they see fit.

** slide 7

The first challenge that I want to present is the one of
reproducibility. On the left hand side, you can see a formalisation of
the SCoPE2 data processing pipeline, as re-implemented in =scp=. The
figure in the middle shows the number of proteins that were identified
using =scp= and those originally published. As you can see, the
disagreement in less than 0.5 percent. On the right hand side, you can
convince yourselves that the quantitative data are also in high
agreement, with only a minority of proteins displaying deviations
from 0.

** slide 8

This slide shows the PCA plot based on the original SCoPE2 paper on
the left, and the one produced with =scp= on the right. Again, you can
appreciate how well these match.

** slide 9

- But what does reproducibility mean? It of course doesn't equate to
  correctness.
- But a study that can be repeated, i.e. one where the results can be
  regenerated from the same input data and the same software
  environment, is one that is coherent.
- A study that can be reproduced, i.e. one where the results can be
  regenerated with new software, is one whose results can be trusted.
- And I'll let you decide what we can conclude from a study that can
  neither be repeated not reproduced.

** slide 10

The second challenge that I want to highlight, and that is omnipresent
in any large scale proteomics experiment, is the one of batch
effects. The PCA figures here are produced with the un-processed
SCoPE2 data, and show that the main effects in the data, those
highlighted along PC1 and PC2, correlate with the chromatography
batches. The biological signal corresponding to the cell-specific
signals are present in the data - within the LC batches along PC1 and
2, or globally, along PC3 and 4.

Batch effects are important but, as seen here for a well designed
experiment, independent of the biological signal, and can thus be
accounted for.

** slide 11

Missing data is also omnipresent in proteomics data, and even more so
in single cell experiments. This figure illustrates
biologically-relevant missingness in the data. On this figure, each
dot represents a proteins. The x and y axis represent the percentage
of missing values in the two cell types. The histograms on the top and
the right confirm that most proteins contain a high proportion of
missing values. Dots are coloured based on the log fold-change between
the two conditions, with red dots indicating higher expression in
macrophages, and blue higher expression in monocytes.

We can see that some proteins are consistently missing more frequently
in one type of cells and we can thus conclude that they are
biologically meaningful.

** slide 12

And we need to be very careful if or when imputing these missing
values. The figure on the right shows the quantitation values of the
VIM protein in all cells, colour-coded by cell type. We see a strong
over-expression in macrophages.

On the left hand side, we have RNF41, a protein with a high percentage
of missing values, a very common case. The grey points are those
obtained after kNN imputation. We immediately see how these imputed
data introduce a serious bias in the data.

** slide 13

And finally, batch effects and missingness aren't independent. The
cells are ordered by date of acquisition and colour-coded based on the
LC batches. We see that the variability in their percentage in missing
data correlates with TMT and LC batches.

Again, nothing that can't be explicitly modelled, but that will
require dedicated methods to be done properly. Indeed, given the
dependence between missingness and batch effects, tackling these
separately in any arbitrary order, is unlikely to provide well
adjusted, and thus biologically optimal data.

** slide 14

With =scp=, we propose to address some specific and general
computational needs in single cell proteomics data analysis. We
propose a principled and standardised tool that supports the needs of
open and reproducible research. The modular pipeline is flexible, and
can adapt multiple use-cases; it isn't limited or specific to the
SCoPE2 data reproduced here. The software comes with ample
documentation and extensive testing. Thanks to its standardised and
modular design, it should favour adoption of the single cell
technologies.

** slide 15

How have we done so far, since =scp= was first released last summer? I
think we did quite well. The interactions with Nikolai Slavov and his
teams helped us initially develop the software. We have since then
contributed some fixed and improvement back, and released a principled
piece of software for data analysis and an environment for further
computational development.

** slide 16

You will find more details about the package at the URLs above. I
would like to conclude by thanking Nikolai and this team for various
discussions, the Bioconductor community for releasing great software,
our funders, and you for your attention and your interest.
