<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Computational challenges in mass spectrometry-based single-cell proteomics</title>
    <meta charset="utf-8" />
    <meta name="author" content="Christophe Vanderaa" />
    <script src="index_files/header-attrs-2.11/header-attrs.js"></script>
    <script src="index_files/xaringanExtra-webcam-0.0.1/webcam.js"></script>
    <script id="xaringanExtra-webcam-options" type="application/json">{"width":"200","height":"200","margin":"1em"}</script>
    <link href="index_files/tile-view-0.2.6/tile-view.css" rel="stylesheet" />
    <script src="index_files/tile-view-0.2.6/tile-view.js"></script>
    <script src="index_files/xaringanExtra_fit-screen-0.2.6/fit-screen.js"></script>
    <link href="index_files/xaringanExtra-extra-styles-0.2.6/xaringanExtra-extra-styles.css" rel="stylesheet" />
    <link href="index_files/panelset-0.2.6/panelset.css" rel="stylesheet" />
    <script src="index_files/panelset-0.2.6/panelset.js"></script>
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
    <link rel="stylesheet" href="custom.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Computational challenges in mass spectrometry-based single-cell proteomics
### Christophe Vanderaa
### 2021/12/03

---








class: middle
name: cc-by

### Get the slides at https://bit.ly/2021_ib2_scp

These slides are available under a **creative common
[CC-BY license](http://creativecommons.org/licenses/by/4.0/)**. You are
free to share (copy and redistribute the material in any medium or
format) and adapt (remix, transform, and build upon the material) for
any purpose, even commercially
&lt;img height="20px" alt="CC-BY" src="https://raw.githubusercontent.com/UCLouvain-CBIO/scp-teaching/main/img/cc1.jpg" /&gt;.

---

class: top

# Single-cell technologies

.panelset[

.panel[.panel-name[Concept]

#### Single-cell technologies unravel cellular heterogeneity

.pull-left[

Subpopulation identification

&lt;img src="./figs/bulk_issue1.svg" width="100%" style="display: block; margin: auto;" /&gt;

]
.pull-right[

Differentiation trajectories

&lt;img src="./figs/bulk_issue2.svg" width="100%" style="display: block; margin: auto;" /&gt;

]
]
.panel[.panel-name[Example]

#### scRNA-Seq on murine embryonic liver

.pull-left[

Subpopulation identification

&lt;img src="./figs/sc_example1.svg" width="100%" style="display: block; margin: auto;" /&gt;

]
.pull-right[

Differentiation trajectories

&lt;img src="./figs/sc_example2.svg" width="100%" style="display: block; margin: auto;" /&gt;

]

Wang et al. 2020. “Comparative Analysis of Cell Lineage Differentiation
during Hepatogenesis in Humans and Mice at the Single-Cell 
Transcriptome Level.” Cell Research 30 (12): 1109–26.

]
]

???

### Concept

Single-cell proteomics is part of the single-cell technologies that are
incredibly powerful when it comes to studying cellular heterogeneity.

The great advantage of single-cell on bulk assays is that they are able
to discover unknown sub-populations, even in well studied tissues. 
Furthermore, single-cell assays capture continuous profiles typical of
differentiation processes. 

Let me show you an example.

### Example

In this study by Wang et al., the authors applied scRNA-Seq on 
embryonic liver. They were able for instance to identify a new 
sub-population of hepatoblasts expressing ID3 that was never described
before. In the same study, they also could reconstruct the hematopoietic
lineage from progenitors that differentiate into neutrophils, 
lymphocytes, dendritic cells,...

But this presentation is about single-cell proteomics, not scRNA-Seq.
So you may ask yourself: why should we bother with proteins?

---

class: top

# RNA vs protein

.panelset[
.panel[.panel-name[Concept]

.left-column[

&lt;img src="./figs/rna_vs_protein_cartoon.svg" width="100%" style="display: block; margin: auto;" /&gt;
]
.right-column[

*RNA &amp;#8776; intention*: RNA is a precursor and "only" provides a proxy
for cell function and identity

*Protein &amp;#8776; function*: proteins bear the cellular function and identity

]
]
.panel[.panel-name[Correlation]

RNA and protein abundances may exhibit poor correlation

&lt;img src="./figs/rna_protein_correlation.svg" width="60%" style="display: block; margin: auto;" /&gt;

Specht et al. 2021. “Single-Cell Proteomic and Transcriptomic Analysis
of Macrophage Heterogeneity Using SCoPE2.” Genome Biology 22 (1): 50.

]
.panel[.panel-name[Measurement]

.pull-left[

&lt;img src="./figs/rna_protein_measurement.svg" width="65%" style="display: block; margin: auto;" /&gt;

]
.pull-right[

#### Proteins are more abundant than RNA

Theoretically, counting RNA copies is more subject to stochasticity
than counting protein copies. 

**Challenge**: improve technology to increase sensitivity and accuracy

#### Protein measurements are computed from multiple peptides 

Estimate protein abundance **AND** variance

**Challenge**: improve computational mapping of peptides to proteins

]
]
]

???

### Concept

Well, let us keep this in mind: RNA represent the intention of a cell.
A cell may or may not commit to it. It is a precursor that is "only"
used as a proxy to infer protein abundance. Protein allows for function.
If the cell would want to rub its face, it would probably build a 
protein for it! Hence the protein content will determine the cell's 
identity. 

But can't we simply say that RNA is equal to protein? No we cannot!

### Correlation

...

Ok it is not the same, but at least RNA at single-cell is much easier
to measure. Yes, for the moment, but something important to 
keep in mind is that scRNA-Seq is possible thanks to PCR amplification.

---

class: top

# Single-cell proteomics experimental workflow

&lt;img src="figs/SCoPE2_method_workflow.svg" width="100%" style="display: block; margin: auto;" /&gt;

---

class: top

# Single-cell proteomics data analysis

&lt;img src="figs/SCoPE2_computation_workflow.svg" width="80%" style="display: block; margin: auto;" /&gt;

---

class: top

# Downstream analyses


.panelset[
.panel[.panel-name[Dimension reduction]

&lt;img src="figs/dimention_reduction.svg" width="30%" style="display: block; margin: auto;" /&gt;

Specht et al. 2021. “Single-Cell Proteomic and Transcriptomic Analysis
of Macrophage Heterogeneity Using SCoPE2.” Genome Biology 22 (1): 50.

]
.panel[.panel-name[Differential abundance]

&lt;img src="figs/differential_abundance.svg" width="40%" style="display: block; margin: auto;" /&gt;

Liang et al. 2020. “Fully Automated Sample Processing and 
Analysis Workflow for Low-Input Proteome Profiling.” Analytical 
Chemistry, December. 

]
.panel[.panel-name[Trajectory analysis]

&lt;img src="figs/trajectory_analysis.svg" width="70%" style="display: block; margin: auto;" /&gt;

Schoof et al. 2021. “Quantitative Single-Cell Proteomics as
a Tool to Characterize Cellular Hierarchies.” Nature Communications 12
(1): 745679.

]
]

---
class: top

# Computational challenges

.panelset[
.panel[.panel-name[Batch effect]

&lt;img src="figs/batch_effect.svg" width="65%" style="display: block; margin: auto;" /&gt;


]
.panel[.panel-name[Missingness]
.pull-left[

#### Imputation leads to artefacts

&lt;img src="figs/imputation_issue.svg" width="100%" style="display: block; margin: auto;" /&gt;

]
.pull-right[

#### Biological missingness

&lt;img src="figs/biological_missingness.svg" width="75%" style="display: block; margin: auto;" /&gt;

]
]
.panel[.panel-name[Batch + Missingness]

&lt;img src="figs/batch_missingness.svg" width="100%" style="display: block; margin: auto;" /&gt;
]
]

---

class: top

# In practice

.panelset[
.panel[.panel-name[Software]
.left-column[

&lt;img src="figs/R.png" width="40%" style="display: block; margin: auto;" /&gt;


&lt;img src="figs/Bioconductor.png" width="80%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;
&lt;br&gt;

&lt;img src="figs/Python.png" width="40%" style="display: block; margin: auto;" /&gt;

]
.right-column[

- `QFeatures` and `scp`: manipulate and visualize single-cell 
  proteomics data
- `scpdata`: collection of published data ready to be analyzed
- SCP.replication: website with replication of published analyses

Vanderaa and Gatto. 2021. “Replication of Single-Cell Proteomics Data
Reveals Important Computational Challenges.” Expert Review of 
Proteomics, October, 1–9.

&lt;br&gt;
&lt;br&gt;

- `SCeptre`: a python alternative to `QFeatures` and `scp`

Schoof et al. 2021. “Quantitative Single-Cell Proteomics as a Tool to
Characterize Cellular Hierarchies.” Nature Communications 12 (1):
745679.

]
]
.panel[.panel-name[scp]


```r
library(scp)
scope2 &lt;- readSCP(featureData = MQdata, colData = annotations, 
                  channelCol = "Channel", batchCol = "Set")
scope2 &lt;- filterFeatures(scope2, ~ Potential.contaminant != "+")
scope2 &lt;- computeSCR(scope2, i = 1:4, colDataCol = "SampleType",
             carrierPattern = "Carrier", samplePattern = "Monocyte")
scope2 &lt;- filterFeatures(scope2, ~ MeanSCR &lt; 0.1)
scope2 &lt;- computeMedianCV(scope2, i = 1:3, proteinCol = "protein", peptideCol = "peptide")
scope2 &lt;- aggregateFeaturesOverAssays(scope2, i = 1:3, name = 4:6, 
                                      fcol = "peptide", fun = robustSummary)
scope2 &lt;- joinAssays(scope2, i = 4:6, name = "peptides")
scope2 &lt;- normalize(scope2, i = "peptides", method = "median", na.rm = TRUE)
scope2 &lt;- logTransform(scope2, i = "normAssay", base = 2)
scope2 &lt;- impute(scope2, i = "normAssay", method = "knn") %&gt;%
scope2 &lt;- aggregateFeatures(i = "logAssay", name = "proteins", fcol = "protein")
```


]
.panel[.panel-name[scpdata]


```r
library(scpdata)
zhu2019EL()
```

```
## An instance of class QFeatures containing 63 assays:
##  [1] 1H1a: SingleCellExperiment with 152 rows and 1 columns 
##  [2] 1H1b: SingleCellExperiment with 267 rows and 1 columns 
##  [3] 1H1c: SingleCellExperiment with 128 rows and 1 columns 
##  ...
##  [61] peptides: SingleCellExperiment with 3444 rows and 60 columns 
##  [62] proteins_intensity: SingleCellExperiment with 840 rows and 60 columns 
##  [63] proteins_iBAQ: SingleCellExperiment with 840 rows and 60 columns
```


]
.panel[.panel-name[SCP.replication]

&lt;img src="figs/scp_replication.png" width="100%" style="display: block; margin: auto;" /&gt;

]
]

---

class: top

# Take home messages

#### Single-cell proteomics is coming!

#### Exciting challenges: model/correct batch effects and missing data

#### Many opportunities: method development, workflow development, benchmarking, ...


Some usefull refs: 

- Slavov, Nikolai. 2021. “Single-Cell Protein Analysis by Mass 
  Spectrometry.” Current Opinion in Chemical Biology 60 (February): 1–9.
- Kelly, Ryan T. 2020. “Single-Cell Proteomics: Progress and 
  Prospects.” Molecular &amp; Cellular Proteomics: MCP 19 (11): 1739–48.
- Vanderaa et Gatto. 2021. “Replication of Single-Cell Proteomics Data
  Reveals Important Computational Challenges.” Expert Review of 
  Proteomics, October, 1–9.

---

class: top

# Thank you for your attention


**Link to the slides**: https://bit.ly/2021_ib2_scp

&lt;br&gt;

&lt;img src="figs/funding.svg" width="40%" style="display: block; margin: auto auto auto 0;" /&gt;

&lt;br&gt;

&lt;img src="figs/get_in_touch.svg" width="100%" style="display: block; margin: auto;" /&gt;
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"ratio": "16:9",
"countIncrementalSlides": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
